packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -1866,7 +1866,13 @@ async function handler(
             });
           }
 
-          if (googleCalResult?.createdEvent?.hangoutLink) {
+          const googleHangoutLink =
+            googleCalResult?.createdEvent?.hangoutLink ??
+            (Array.isArray(googleCalResult?.updatedEvent)
+              ? googleCalResult.updatedEvent[0]?.hangoutLink
+              : googleCalResult?.updatedEvent?.hangoutLink);
```

Comment:
AFAIU, the issue occurred because the `originalEvent` didn't have videoCallData(because it was confirmed later and then the videoCallData isn't updated I guess). 

The thing is that we should have used the latest link returned by the updateMeeting/updateEvent call(which is available in `updatedEvent`). We can fallback to `createdEvent` just in case. It ensure that the link expiry is up-to-date(and maybe other details).

Though I tested, and it makes sense that, during reschedule `createdEvent` won't be there. So, technically we should just use `updatedEvent`.

Reply:
> Though I tested, and it makes sense that, during reschedule createdEvent won't be there. So, technically we should just use updatedEvent.

I also thought about only using updatedEvent but when you reschedule and add a new guest in booking form then you would get createdEvent.

<img width="1135" alt="Screenshot 2024-05-04 at 7 20 55 PM" src="https://github.com/calcom/cal.com/assets/53316345/b8c52762-e6ec-42ca-8d9f-a72bbf35c408">

Reply:
Ah cool that's something i missed. But we should still prioritize `updatedEvent` over `createdEvent`. Not sure if that change is done already.